src/Agent.Worker/Build/GitSourceProvider.cs

```diff
@@ -25,18 +25,26 @@ public class GitSourceProvider : SourceProvider, ISourceProvider
 
         public override string RepositoryType => WellKnownRepositoryTypes.Git;
 
-        public virtual async Task GetSourceAsync(IExecutionContext executionContext, ServiceEndpoint endpoint, CancellationToken cancellationToken)
+        public virtual async Task GetSourceAsync(
+            IExecutionContext executionContext,
+            ServiceEndpoint endpoint,
+            CancellationToken cancellationToken)
         {
             Trace.Entering();
+            // Validate args.
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
             ArgUtil.NotNull(endpoint, nameof(endpoint));
 
             executionContext.Output($"Syncing repository: {endpoint.Name} (Git)");
             _gitCommandManager = HostContext.GetService<IGitCommandManager>();
             await _gitCommandManager.LoadGitExecutionInfo(executionContext);
 
-            string targetPath = executionContext.Variables.Get(Constants.Variables.Build.SourcesDirectory);
-            string sourceBranch = executionContext.Variables.Get(Constants.Variables.Build.SourceBranch);
-            string sourceVersion = executionContext.Variables.Get(Constants.Variables.Build.SourceVersion);
+            string targetPath;
+            string sourceBranch;
+            string sourceVersion;
+            endpoint.Data.TryGetValue(Constants.Variables.Build.SourcesDirectory, out targetPath);
```

Comment:
I'm not advocating "dont-repeat-yourself violation". I don't think reusing build variable names in the endpoint dictionary makes sense - e.g. "Build.SourceVersion". See my example above, it's just "SourceVersion".

It's not technically different, I just think it's less confusing - a readability concern.